test-ng: test_groups.py does not include `expexted_users`

**What happened:**

After the merge of https://github.com/gardenlinux/gardenlinux/pull/3808, I see this when running the tests locally in qemu. Cloud tests could also be affected.

I guess we need add `expected_users` (passed on cmdline) to the tests.


```
❯ ./test-ng --ssh --skip-cleanup .build/aws-gardener_prod-amd64-today-b690532b.raw

...

🧪  running tests with args: ./run_tests --junit-xml=/dev/virtio-ports/test_junit --metadata Artifact aws-gardener_prod-amd64-today-b690532b.raw --metadata Type qemu --metadata Namespace test-ng --system-booted --allow-system-modifications --expected-users gardenlinux

...

======================================================= FAILURES =======================================================
____________________________________________ test_group_wheel_has_no_users _____________________________________________
test_groups.py:56: in test_group_wheel_has_no_users
    assert (
E   AssertionError: user group wheel is not empty as expected. Was: ['gardenlinux']
E   assert 1 == 0
E    +  where 1 = len(['gardenlinux'])
E    +    where ['gardenlinux'] = Group(groupname='wheel', password='x', gid='105', user_list=['gardenlinux']).user_list

```

--- a/tests-ng/test_groups.py
+++ b/tests-ng/test_groups.py
@@ -1,4 +1,4 @@
-from typing import List
+from typing import List, Set
 
 import pytest
 from plugins.linux_etc_files import Group
@@ -44,7 +44,9 @@ def test_group_root_has_no_users(group_entries: List[Group]):
 
 
 @pytest.mark.feature("not _dev and not pythonDev and not container")
-def test_group_wheel_has_no_users(group_entries: List[Group]):
+def test_group_wheel_has_no_unexpected_users(
+    group_entries: List[Group], expected_users: Set[str]
+):
     """
     Check if parameterized groups have the expected users.
     """
@@ -53,6 +55,11 @@ def test_group_wheel_has_no_users(group_entries: List[Group]):
 
     assert group, "Group wheel is not present."
 
+    # remove expected users from group as we want to check for unexpected users
+    expected_user_list = [
+        user for user in group.user_list if user not in expected_users
+    ]
+
     assert (
-        len(group.user_list) == 0
-    ), f"user group wheel is not empty as expected. Was: {group.user_list}"
+        len(expected_user_list) == 0
+    ), f"user group wheel is not empty as expected. Was: {expected_user_list}"

